### ISSUE TITLE
modprobe.d/systemd.conf is not loaded

### ISSUE THREAD

**Issue by gravndal** [NixOS/nixpkgs#151771]
## Issue description

The default [systemd.conf](https://github.com/systemd/systemd/blob/main/modprobe.d/systemd.conf) sets `options bonding max_bonds=0` due to the following reason as given in the file: 

> When bonding module is loaded, it creates bond0 by default due to max_bonds option default value 1. This interferes with the network configuration management / networkd, as it is not possible to detect whether this bond0 was intentionally configured by the user, or should be managed by networkd/NM/etc. Therefore disable bond0 creation.

Without this, configuring `bond0` with systemd-networkd is (for some options like `mode` at least) essentially a NOOP. This is rather confusing, when `systemd-analyze cat-config modprode.d` indicates that it is being loaded (which is an issue in itself I suppose).

Reproduces with the following `flake.nix`, `flake.lock`, and `nixos-rebuild build-vm --flake [...]`:
```
{
  inputs = {
    nixpkgs = { url = "github:nixos/nixpkgs/master"; };
  };

  outputs = { self, nixpkgs, ... }: {
    nixosConfigurations = {
      repro = nixpkgs.lib.nixosSystem {
        system = "x86_64-linux";
        modules = [
          ({ ... }: {
            users.users.a = {
              password = "a";
              isNormalUser = true;
            };
            networking = {
              hostName = "repro";
              useDHCP = false;
              useNetworkd = true;
            };
            systemd.network = {
              enable = true;
              netdevs = {
                "50-bond0" = {
                  netdevConfig = {
                    Name = "bond0";
                    Kind = "bond";
                  };
                  bondConfig = {
                    Mode = "active-backup";
                  };
                };
              };
            };
          })
        ];
      };
    };
  };
}
```
```
{
  "nodes": {
    "nixpkgs": {
      "locked": {
        "lastModified": 1640182329,
        "narHash": "sha256-tMCOM9n1RsTit3iLY1Oi0VLgXA+pvJWc9Rv8HuZz6mU=",
        "owner": "nixos",
        "repo": "nixpkgs",
        "rev": "4c252f40af8391b4f5efec91db9c6daf1818bd45",
        "type": "github"
      },
      "original": {
        "owner": "nixos",
        "ref": "master",
        "repo": "nixpkgs",
        "type": "github"
      }
    },
    "root": {
      "inputs": {
        "nixpkgs": "nixpkgs"
      }
    }
  },
  "root": "root",
  "version": 7
}
```

This produces the following unexpected configuration:
```
$ grep 'Bonding Mode' /proc/net/bonding/bond0
Bonding Mode: load balancing (round-robin)
```

When the expected outcome obtained after setting:
```
boot.extraModprobeConfig = ''
  options bonding max_bonds=0
'';
```
is:
```
$ grep 'Bonding Mode' /proc/net/bonding/bond0
Bonding Mode: fault-tolerance (active-backup)
```

CC systemd maintainers:  @andir @edolstra @flokli @Kloenk

**Comment by flokli** [comment]
Maybe we should add this to `nixos/modules/system/boot/modprobe.nix` and `nixos/modules/system/boot/stage-1.nix`.

cc @arivanvp

**Comment by Luflosi** [comment]
This has been implemented in #157839.
